hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -1102,6 +1152,50 @@ long generateDiffReport(final String jobId,
     }
   }
 
+  private boolean isBlockLocationSame(
+      String fromObjectName,
+      String toObjectName,
+      final Table<String, ? extends WithObjectID> fromSnapshotTable,
+      final Table<String, ? extends WithObjectID> toSnapshotTable
+  ) throws IOException {
+
+    if (fromObjectName == null && toObjectName == null) {
+      LOG.debug("fromObjectName and toObjectName are null.");
+      return true;
+    }
+
+    if (fromObjectName == null || toObjectName == null) {
+      LOG.debug("fromObjectName: '{}' or toObjectName: '{}' is null.",
+          fromObjectName, toObjectName);
+      return false;
+    }
+
+    final WithObjectID fromObject = fromSnapshotTable.get(fromObjectName);
+    final WithObjectID toObject = toSnapshotTable.get(toObjectName);
+
+    if (fromObject == null && toObject == null) {
+      LOG.debug("fromObject and toObject both are null.");
+      return true;
+    }
+
+    if (fromObject == null || toObject == null) {
+      LOG.debug("fromObject: '{}' or toObject: '{}' is null.",
+          fromObject, toObject);
+      return false;
+    }
+
+    if (!(fromObject instanceof OmKeyInfo) ||
+        !(toObject instanceof OmKeyInfo)) {
+      LOG.debug("fromObject is instance of '{}' and toObject is of '{}'.",
+          fromObject.getClass().getSimpleName(),
+          toObject.getClass().getSimpleName());
+      return false;
+    }
+
+    return SnapshotDeletingService.isBlockLocationInfoSame(
```

Comment:
We can directly call this function for fromObject & toObject. We need not check the above if cases.

Reply:
Initially integration tests were failing so I added that.  I removed the redundant null checks from here and change it to exception because we should not be in the situation.